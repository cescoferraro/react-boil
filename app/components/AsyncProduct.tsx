import * as React from "react"
import { asyncComponent } from 'react-async-component'

export default asyncComponent({
    resolve: () => Promise.resolve().then(() => require("./Product.tsx"))
});
