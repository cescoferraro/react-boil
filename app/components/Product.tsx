import * as React from "react"
declare var System: any
import * as CSS from "../css/teste.css"

export default () => (
    <div className={CSS.test}>
        <h2>I am on a separate Bundle!</h2>
    </div>
)
