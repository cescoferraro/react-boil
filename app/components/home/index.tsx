import * as React from "react"
import * as CSS from "./css/home.css"
const test = require("../../../README.org")

export const HomeComponent = ({ }) => {
    return <div className={CSS.container} dangerouslySetInnerHTML={{ __html: test }} />
}
