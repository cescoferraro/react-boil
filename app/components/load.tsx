import universal from "react-universal-component"

export const UniversalComponent = universal(() => import("./product"), {
    resolve: () => require.resolveWeak("./product"),
    chunkName: "Universla",
    onError: (erros) => {
        console.log(erros)
    }
})
