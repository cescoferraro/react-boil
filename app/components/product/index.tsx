import universal from "react-universal-component"
declare var System: any

export const UniversalComponent = universal(
    () => System.import(/* webpackChunkName: 'Product' */ "./Product"),
    {
        resolve: () => require.resolveWeak("./Product"),
        chunkName: "Product"
    }
)
