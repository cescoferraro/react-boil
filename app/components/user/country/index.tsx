import universal from "react-universal-component"
declare var System: any

export const AsyncCountry = universal(
    () => System.import(/* webpackChunkName: 'country' */ "./country"),
    {
        chunkName: "country"
    }
)
