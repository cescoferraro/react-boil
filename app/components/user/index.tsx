import universal from "react-universal-component"
declare var System: any
import * as importCss from 'babel-plugin-universal-import/importCss.js'
import * as universalImport from 'babel-plugin-universal-import/universalImport.js'


declare var __dirname: any
export const AsyncUser = universal(
    universalImport({
        load: () => Promise.all([
            System.import(/* webpackChunkName: 'user' */ "./user"),
            importCss("user")]).then(promises => promises[0]),
        chunkName: () => "user",
        resolve: () => require.resolveWeak('./user'),
    })
)

