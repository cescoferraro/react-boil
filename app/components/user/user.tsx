import * as React from "react"
import * as CSS from "./css/teste.css"
import { ListSubheader as Subheader } from 'material-ui/List'
import { ProfileClass } from "../../../store/reducers"
import Divider from "material-ui/Divider"
import { Item } from "./item/item"
import { compose } from "recompose"
import Code from "material-ui-icons/Code.js"
import sizeMe from "react-sizeme"
import Country from "./country/country"
import Terrain from "material-ui-icons/Terrain.js"
import LocationCity from "material-ui-icons/LocationCity.js"
import AccountCircle from "material-ui-icons/AccountCircle.js"
import GpsFixed from "material-ui-icons/GpsFixed.js"




import { withStyles, createStyleSheet } from 'material-ui/styles';
import Card, { CardActions, CardContent } from 'material-ui/Card';
import Button from 'material-ui/Button';
import Typography from 'material-ui/Typography';

const styleSheet = createStyleSheet(theme => ({
    card: {
        minWidth: 275,
    },
    bullet: {
        display: 'inline-block',
        margin: '0 2px',
        transform: 'scale(0.8)',
    },
    title: {
        marginBottom: 16,
        fontSize: 14,
        color: theme.palette.text.secondary,
    },
    pos: {
        marginBottom: 12,
        color: theme.palette.text.secondary,
    },
}));

export class UserComponent extends React.Component<any, any> {
    public render() {
        const profile: ProfileClass = new ProfileClass(this.props.profiles[this.props.profile.id])
        const classes = this.props.classes;
        const bull = <span className={classes.bullet}>â€¢</span>;
        const grey = { fill: "grey" }
        return (
            <div className={CSS.test}>
                <Card className={classes.card}>
                    <CardContent>

                        <div className={CSS.title}>
                            <img
                                className={CSS.image}
                                alt=""
                                src={this.props.size.width > 300 ? profile.picture.large : profile.picture.medium}
                                width={150}
                            />
                            <div className={CSS.name}>
                                <a>{profile.name.first}</a>
                            </div>
                        </div>
                        <Item
                            title="Id"
                            value={profile.id}
                            icon={<AccountCircle style={grey} />}
                        />
                        <Item
                            title="Fullname"
                            value={profile.fullname()}
                            icon={<AccountCircle style={grey} />}
                        />
                        <Item
                            title="Email"
                            value={profile.email}
                            icon={<GpsFixed style={grey} />}
                        />
                        <Subheader>Address</Subheader>
                        <Item
                            title="Street"
                            value={profile.location.street}
                            icon={<Code style={grey} className={CSS.flag} />}
                        />
                        <Item
                            title="City"
                            value={profile.location.city}
                            icon={<LocationCity style={grey} className={CSS.flag} />}
                        />
                        <Item
                            title="Zip-Code"
                            value={profile.location.postcode}
                            icon={<Code style={grey} className={CSS.flag} />}
                        />
                        <Item
                            title="State"
                            value={profile.location.state}
                            icon={<Terrain style={grey} className={CSS.flag} />}
                        />
                        <Country profile={profile} className={CSS.flag} />
                    </CardContent>
                    <CardActions>
                        <Button dense>Learn More</Button>
                    </CardActions>
                </Card>
            </div>
        )
    }
}

export default compose(sizeMe(), withStyles(styleSheet))(UserComponent)
