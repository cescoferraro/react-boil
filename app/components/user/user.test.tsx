import * as React from "react"
import { UserComponent } from "./user"
import { profileStartup } from "../../../store/reducers"
import { MUIProvider } from "../../../shared/components/MUIProvider"

test("Render a User", () => {
    const component = render(
        <MUIProvider>
            <UserComponent
                profiles={{ 123: profileStartup }}
                size={{ width: 300, height: 600 }}
                profile={profileStartup}
            />
        </MUIProvider>
    )
    expect(component).toMatchSnapshot()
})
