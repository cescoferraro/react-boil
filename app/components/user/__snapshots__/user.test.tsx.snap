// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render a User 1`] = `
<div
  class="test"
>
  <div
    style="color:rgba(0, 0, 0, 0.87);background-color:#ffffff;transition:all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms;box-sizing:border-box;font-family:Roboto, sans-serif;-webkit-tap-highlight-color:rgba(0,0,0,0);box-shadow:0 1px 6px rgba(0, 0, 0, 0.12),
           0 1px 4px rgba(0, 0, 0, 0.12);border-radius:2px;z-index:1;mui-prepared:;"
  >
    <div
      style="padding-bottom:0;"
    >
      <div
        class="title"
      >
        <img
          alt=""
          class="image"
          src="http://via.placeholder.com/300x300"
          width="150"
        />
        <div
          class="name"
        >
          <a>
            John
          </a>
        </div>
      </div>
      <div
        class="item"
      >
        <div
          class="itemIcon"
        >
          <svg
            style="display:inline-block;color:rgba(0, 0, 0, 0.87);fill:grey;height:24px;width:24px;user-select:none;transition:all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms;mui-prepared:;"
            viewbox="0 0 24 24"
          >
            <path
              d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 3c1.66 0 3 1.34 3 3s-1.34 3-3 3-3-1.34-3-3 1.34-3 3-3zm0 14.2c-2.5 0-4.71-1.28-6-3.22.03-1.99 4-3.08 6-3.08 1.99 0 5.97 1.09 6 3.08-1.29 1.94-3.5 3.22-6 3.22z"
            />
          </svg>
        </div>
        <div
          class="itemText"
        >
          <a
            class="key"
          >
             Fullname 
          </a>
          <a
            class="value"
          >
             Mr. John Doe 
          </a>
        </div>
      </div>
      <div
        class="item"
      >
        <div
          class="itemIcon"
        >
          <svg
            style="display:inline-block;color:rgba(0, 0, 0, 0.87);fill:grey;height:24px;width:24px;user-select:none;transition:all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms;mui-prepared:;"
            viewbox="0 0 24 24"
          >
            <path
              d="M12 8c-2.21 0-4 1.79-4 4s1.79 4 4 4 4-1.79 4-4-1.79-4-4-4zm8.94 3c-.46-4.17-3.77-7.48-7.94-7.94V1h-2v2.06C6.83 3.52 3.52 6.83 3.06 11H1v2h2.06c.46 4.17 3.77 7.48 7.94 7.94V23h2v-2.06c4.17-.46 7.48-3.77 7.94-7.94H23v-2h-2.06zM12 19c-3.87 0-7-3.13-7-7s3.13-7 7-7 7 3.13 7 7-3.13 7-7 7z"
            />
          </svg>
        </div>
        <div
          class="itemText"
        >
          <a
            class="key"
          >
             Email 
          </a>
          <a
            class="value"
          >
             gabriel.mathieu@example.com 
          </a>
        </div>
      </div>
      <div
        style="box-sizing:border-box;color:rgba(0, 0, 0, 0.54);font-size:14px;font-weight:500;line-height:48px;padding-left:16px;width:100%;mui-prepared:;"
      >
        Address
      </div>
      <div
        class="item"
      >
        <div
          class="itemIcon"
        >
          <svg
            class="flag"
            style="display:inline-block;color:rgba(0, 0, 0, 0.87);fill:grey;height:24px;width:24px;user-select:none;transition:all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms;mui-prepared:;"
            viewbox="0 0 24 24"
          >
            <path
              d="M9.4 16.6L4.8 12l4.6-4.6L8 6l-6 6 6 6 1.4-1.4zm5.2 0l4.6-4.6-4.6-4.6L16 6l6 6-6 6-1.4-1.4z"
            />
          </svg>
        </div>
        <div
          class="itemText"
        >
          <a
            class="key"
          >
             Street 
          </a>
          <a
            class="value"
          >
             2976 quai charles-de-gaulle 
          </a>
        </div>
      </div>
      <div
        class="item"
      >
        <div
          class="itemIcon"
        >
          <svg
            class="flag"
            style="display:inline-block;color:rgba(0, 0, 0, 0.87);fill:grey;height:24px;width:24px;user-select:none;transition:all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms;mui-prepared:;"
            viewbox="0 0 24 24"
          >
            <path
              d="M15 11V5l-3-3-3 3v2H3v14h18V11h-6zm-8 8H5v-2h2v2zm0-4H5v-2h2v2zm0-4H5V9h2v2zm6 8h-2v-2h2v2zm0-4h-2v-2h2v2zm0-4h-2V9h2v2zm0-4h-2V5h2v2zm6 12h-2v-2h2v2zm0-4h-2v-2h2v2z"
            />
          </svg>
        </div>
        <div
          class="itemText"
        >
          <a
            class="key"
          >
             City 
          </a>
          <a
            class="value"
          >
             pully 
          </a>
        </div>
      </div>
      <div
        class="item"
      >
        <div
          class="itemIcon"
        >
          <svg
            class="flag"
            style="display:inline-block;color:rgba(0, 0, 0, 0.87);fill:grey;height:24px;width:24px;user-select:none;transition:all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms;mui-prepared:;"
            viewbox="0 0 24 24"
          >
            <path
              d="M9.4 16.6L4.8 12l4.6-4.6L8 6l-6 6 6 6 1.4-1.4zm5.2 0l4.6-4.6-4.6-4.6L16 6l6 6-6 6-1.4-1.4z"
            />
          </svg>
        </div>
        <div
          class="itemText"
        >
          <a
            class="key"
          >
             Zip-Code 
          </a>
          <a
            class="value"
          >
             8831 
          </a>
        </div>
      </div>
      <div
        class="item"
      >
        <div
          class="itemIcon"
        >
          <svg
            class="flag"
            style="display:inline-block;color:rgba(0, 0, 0, 0.87);fill:grey;height:24px;width:24px;user-select:none;transition:all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms;mui-prepared:;"
            viewbox="0 0 24 24"
          >
            <path
              d="M14 6l-3.75 5 2.85 3.8-1.6 1.2C9.81 13.75 7 10 7 10l-6 8h22L14 6z"
            />
          </svg>
        </div>
        <div
          class="itemText"
        >
          <a
            class="key"
          >
             State 
          </a>
          <a
            class="value"
          >
             graub√ºnden 
          </a>
        </div>
      </div>
      <div
        class="item"
      >
        <div
          class="itemIcon"
        >
          <img
            alt="Canada Flag"
            class="flag"
            src="/flags/flags-iso/flat/32/BR.png"
          />
        </div>
        <div
          class="itemText"
        >
          <a
            class="key"
          >
             Nationality 
          </a>
          <a
            class="value"
          >
             BR 
          </a>
        </div>
      </div>
      <hr
        style="margin:0;margin-top:-1px;margin-left:0;height:1px;border:none;background-color:#e0e0e0;mui-prepared:;"
      />
    </div>
  </div>
</div>
`;
